src/doc/using_visit/Quantitative/Expressions.rst

```diff
@@ -2253,5 +2254,14 @@ Just as for centering, we have two options when dealing with variables from
 two different meshes. Each of which involves *mapping* one of the variables
 onto the other variable's mesh using one of the CMFE expression functions.
 
+Deferring expression evaluation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Expressions are generally evaluated before any operators are applied to the data.
+There are instances where one may want to defer the evaluation until after other filters have been applied.
```

Comment:
```suggestion
In cases where an expression involves the *output* of an operator, it is necessary to *defer* evaluation of such an expression to *after* operators have been applied.
```

Reply:
FWIW, I think the intro motivating use of **DeferExpression** operator in intro here,

https://github.com/visit-dav/visit/blob/4380e48962812444c93e1f1c4b6a2b2bdafeef8c/src/doc/using_visit/Operators/OperatorTypes/DeferExpression_operator.rst?#L5-L10

could use similar statements.

Reply:
Re: your suggestion, how would you add wording about operators that change the outcome of an expression, eg an operator that removes cells or points could affect some expressions, correct?

Reply:
Maybe I am confused but I liked the explanation you included for when to use the **Defer_Expression** operator.

As an aside, I don't think the choice to use **Defer Expression** operator is so much a user preference as it is a *necessary* thing to do for certain expressions...those involving variables computed by operators.

So, I thought it would help to include that explanation and motivation in the introductory part of this file too.

My read was that both the intro and your explanation are serving the same purpose...to motiviate use of the **Defer Expression** operator.

Did I misread?

Reply:
Didn't mean to confuse you. I was just thinking about perhaps a volume expression and  transform-scale operator that scaled the mesh and changed the volume of the cells. The volume expression isn't using an operator-created quantity, but an operator-modified quantity.  You could opt *not* to defer the volume expression if you wanted to see the original volumes.  Have no idea if this use-case even makes sense though.

Reply:
Oh, wow! I hadn't realized that kind of scenario existed. So, there *are* cases where the decision to defer is indeed a *choice* depending on what user intended.

Now, since *expressions* are kinda sorta *always* treated like any other database variable in that they are created *before* entering any operators, I think the user's intent is captured at the moment they choose to add the *scale* operator. If they've added the scale operator, then its my view that they want scaled volumes. If they don't want volume to be scaled, the won't add the scale operator. Its harder for me to imagine a use case where they want something scaled by the scale operator but then decide they want to see something that isn't effected by the operator. In this way, it kinda sorta feels like the question of *original* or *actual* we have in a number of queries.

Reply:
It just now occurs to me too...If all operators had been designed from the start to work *both* as an operator and as an expression, then **Defer Expression** wouldn't have even been needed, right? So, the fact that we've created **Defer Expression** has a lot to do with how the code has evolved. Is that a fair assessement?